com.rustdesk.RustDesk.json

```diff
@@ -0,0 +1,91 @@
+            "sha256": "eff47a4ecd833fbf18de9686632a70ee8d0794b79aecb217ebd0ce11db4cd0db"
+          }
+        ]
+      },
+      {
+        "name": "libarchive",
+        "buildsystem": "autotools",
+        "config-opts": ["--disable-static"],
+        "cleanup": ["/include", "/lib/*.la", "/lib/pkgconfig", "/share"],
+        "sources": [
+            {
+                "type": "archive",
+                "url": "https://github.com/libarchive/libarchive/releases/download/v3.5.1/libarchive-3.5.1.tar.gz",
+                "sha256": "9015d109ec00bb9ae1a384b172bf2fc1dff41e2c66e5a9eeddf933af9db37f5a"
+            }
+        ]
+      },
+      {
+        "name": "rustdesk",
+        "buildsystem": "simple",
+        "build-commands": [
+          "/app/bin/bsdtar -zxvf rustdesk*.deb",
+          "rm -rf /app/bin",
+          "tar -xvf ./data.tar.xz",
+          "cp -r usr/*  /app/",
+          "mkdir -p /app/bin && ln -s /app/lib/rustdesk/rustdesk /app/bin/rustdesk",
+          "mv /app/share/applications/rustdesk.desktop /app/share/applications/com.rustdesk.RustDesk.desktop",
+          "mv /app/share/applications/rustdesk-link.desktop /app/share/applications/com.rustdesk.RustDesk-link.desktop",
+          "sed -i '/^Icon=/ c\\Icon=com.rustdesk.RustDesk' /app/share/applications/*.desktop",
+          "install -Dm644 usr/share/icons/hicolor/256x256/apps/rustdesk.png /app/share/icons/hicolor/256x256/apps/com.rustdesk.RustDesk.png"
+        ],
+        "cleanup": ["/include", "/lib/pkgconfig", "/share/gtk-doc", "/share/man"],
+        "sources": [
+          {
+            "type": "file",
+            "only-arches": ["aarch64"],
+            "url": "https://github.com/rustdesk/rustdesk/releases/download/1.2.4/rustdesk-1.2.4-aarch64.deb",
+            "sha256": "4f8be7199a1e8acf0799ce8cea6af08bcf2d6f073a3a098809ac842e3a87d1e8",
+            "x-checker-data": {
+              "type": "anitya",
+              "project-id": "309593",
+              "url-template": "https://github.com/rustdesk/rustdesk/releases/download/$version/rustdesk-$version-aarch64.deb"
+            }
+          },
+          {
+            "type": "file",
+            "only-arches": ["x86_64"],
+            "url": "https://github.com/rustdesk/rustdesk/releases/download/1.2.4/rustdesk-1.2.4-x86_64.deb",
+            "sha256": "43c2f43ad7a8a173fa51093944243f04f003f675e74812644780eee1468b3f9b",
+            "x-checker-data": {
+              "type": "anitya",
+              "project-id": "309593",
+              "url-template": "https://github.com/rustdesk/rustdesk/releases/download/$version/rustdesk-$version-x86_64.deb"
+            }
+          }
+        ]
+      }
+    ],
+    "finish-args": [
+      "--share=ipc",
+      "--socket=x11",
+      "--socket=fallback-x11",
+      "--socket=wayland",
+      "--share=network",
+      "--filesystem=home",
+      "--device=dri",
+      "--socket=pulseaudio",
+      "--talk-name=org.freedesktop.Flatpak"
```

Comment:
This is remote desktop software, these are required to run it. Please check anydesk https://github.com/flathub/com.anydesk.Anydesk/blob/master/com.anydesk.Anydesk.json

Reply:
I'm going to need more than that.

Reply:
Also this is not allowed by default, you are going to need an exception. Otherwise it won't build.

Reply:
Maybe it is required to make this feature works?
Or if have any other methods that have smaller permission and won't interfere this feature?

https://github.com/rustdesk/rustdesk/pull/7717

Reply:
Again

Below two remote desktop (we are the same App).
https://github.com/flathub/net.christianbeier.MultiVNC/blob/master/net.christianbeier.MultiVNC.yml#L12

https://github.com/flathub/com.anydesk.Anydesk/blob/635ebbbdb4185c5edeba387dad864a0612c54976/com.anydesk.Anydesk.json#L18

Reply:
> Below two remote desktop on flathub are good examples.

You do realise that none of the manifests you show as examples does not actually use `org.freedesktop.Flatpak`. 

Again what do you need to run on host?

Reply:
> Maybe it is required to make this feature works?
Or if have any other methods that have smaller permission and won't interfere this feature?
> https://github.com/rustdesk/rustdesk/pull/7717

Why would a feature related to cursors require executing arbitrary commands on host?

Reply:
Please check this, https://github.com/rustdesk/rustdesk/discussions/1537#discussioncomment-3782994

Reply:
AnyDesk does need to use this because it support x11 only, but we support x11 and wayland both, we need this feature.

Reply:
We add these features not for playing, it is because we really need.

Reply:
> > Maybe it is required to make this feature works?
> > Or if have any other methods that have smaller permission and won't interfere this feature?
> > [rustdesk/rustdesk#7717](https://github.com/rustdesk/rustdesk/pull/7717)
> 
> Why would a feature related to cursors require executing arbitrary commands on host?

Ignore this.

Reply:
Ok that makes sense. Thanks. However I see fallbacks everywhere in case loginctl does not work. Is that not enough? 

Eg here https://github.com/rustdesk/rustdesk/blob/9827c7651442800f47428087903d52da8cb8c2c8/libs/hbb_common/src/platform/linux.rs#L115-L123